lib/game.rb

```diff
@@ -1,27 +1,43 @@
 require_relative './player.rb'
 require_relative './board.rb'
 require_relative './ui.rb'
+require 'set'
 
 class Game
   include UI
-  def initialize(board,players)
+  def initialize(board, players)
     @board = board
     @players = players
+  end
+
+  def play
     output('Welcome to Tic-tac-toe!', true)
     2.times { |n| get_player_data(n) }
     @current_player = @players[0]
     @other_player = @players[1]
-    while !won? || !full?
+    loop do
       clear_console
       output(@players[0].player_info)
-      output(@players[1].player_info,true)
-      grid=@board.get_grid
-      output(grid,true)
-      ask_move(@current_player)
-      switch_players()
+      output(@players[1].player_info, true)
+      grid = @board.get_grid
+      output(grid, true)
+      if won_by
+        output("#{@players[won_by].name} won the game!", true)
+        break
+      end
+      if full?
+        output('The game ended with a draw!', true)
+        break
+      end
+      move = ask_move(@current_player)
+      @current_player.add_position(move)
+      @board.update(move, @current_player.sign)
+      switch_players
     end
   end
 
+  private
+
   def get_player_data(n)
```

Comment:
Shouldn't this method of getting player name be in the Player class rather than here in the Game class?

Reply:
Moved it to the Player class.